pkg/asset/machines/machineconfig/powersmt.go

```diff
@@ -0,0 +1,70 @@
+package machineconfig
+
+import (
+	"fmt"
+
+	ignutil "github.com/coreos/ignition/v2/config/util"
+	igntypes "github.com/coreos/ignition/v2/config/v3_2/types"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/installer/pkg/asset/ignition"
+	mcfgv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+)
+
+// ForPowerSMT sets the SMT level for Power machines.
+func ForPowerSMT(role string, smtLevel string) (*mcfgv1.MachineConfig, error) {
+	powerSMTUnit, err := createPowerSMTUnit(smtLevel)
+	if err != nil {
+		return nil, err
+	}
+
+	ignConfig := igntypes.Config{
+		Ignition: igntypes.Ignition{
+			Version: igntypes.MaxVersion.String(),
+		},
+		Systemd: igntypes.Systemd{
+			Units: []igntypes.Unit{
+				{
+					Contents: &powerSMTUnit,
+					Name:     fmt.Sprintf("99-%s-powersmt.service", role),
+					Enabled:  ignutil.BoolToPtr(true),
+				},
+			},
+		},
+	}
+
+	rawExt, err := ignition.ConvertToRawExtension(ignConfig)
```

Comment:
I know in the MCO we also just json.Marshal ignition usually rather than `ConvertToRawExtension` might be worth writing some tests here to ensure this works?

Reply:
All other machineconfigs in the installer do this, but I could add test in a follow up if you'd like!